### ISSUE TITLE
Subcommand description in top level help message

### ISSUE THREAD

**Issue by AgamAgarwal** [druths/arghandler#9]
The current format of the top-level help message is this:

```
usage: myprog [-h] {foo,bar} ...

positional arguments:
  {foo,bar}
  cargs       arguments for the subcommand

optional arguments:
  -h, --help  show this help message and exit
```

I think that the following would be more helpful:

```
usage: myprog [-h] {foo,bar} ...

positional arguments:
  foo       Does foo
  bar       Does bar

optional arguments:
  -h, --help  show this help message and exit
```

(Also, what is the use of the option `cargs` in the help text)

A possible way to use this feature can be:

``` python
@subcmd(help="Does foo")
def foo(parser, context, args):
    print "Doing foo"
```

**Comment by druths** [comment]
I really like this idea.  Looking into it, though, getting this behavior is less than trivial.  Arghandler is built on top of argparse.  It turns out that argparse doesn’t have a transparent way to replace some of the help message.  It’s possible through a hack to replace the whole help message, but this means that we’ll have to implement all the code for correctly handling the printing of the options (which is do-able, but non-trivial to get right).

Any ideas?  I don’t have time to reimplement this capability right now – but do think that it should be done.  If it’s up to me, I’ll leave it for a later version.

**Comment by AgamAgarwal** [comment]
Solved in pull request #10
